src/server/game/Entities/Player/Player.cpp

```diff
@@ -11722,7 +11722,7 @@ void Player::SendInstanceResetWarning(uint32 mapid, Difficulty difficulty, uint3
 
 void Player::ApplyEquipCooldown(Item* pItem)
 {
-    if (pItem->HasFlag(ITEM_FIELD_FLAGS, ITEM_FLAG_NO_EQUIP_COOLDOWN))
+    if (pItem->GetTemplate()->HasFlag(ITEM_FLAG_NO_EQUIP_COOLDOWN))
```

Comment:
Fixes the issue where ITEM_FLAG_NO_EQUIP_COOLDOWN was never caught as previously it was checking the DB flag in item_field_flags.